torch/distributions.py

```diff
@@ -210,20 +214,22 @@ class Gamma(Distribution):
     """
 
     def __init__(self, alpha, beta):
+        if isinstance(alpha, Number):
+            if isinstance(beta, Number):
+                alpha = torch.Tensor([alpha])
+                beta = torch.Tensor([beta])
+            else:
+                alpha = type(beta)(*beta.size()).fill_(alpha)
+        elif isinstance(beta, Number):
+            beta = type(alpha)(*alpha.size()).fill_(beta)
```

Comment:
This might return a result on a different GPU than `alpha`. Use `alpha.new().resize_as_(alpha).fill_(beta)`

Reply:
BTW I think it might be rewritten in a bit clearer way:
```python
alpha_num = isinstance(alpha, Number)
beta_num = isinstance(beta, Number)
if alpha_num and not beta_num:
    alpha = beta.new(beta.size()).fill_(alpha)
elif not alpha_num and beta_num:
    beta = alpha.new(alpha.size()).fill_(beta)
elif alpha_num and beta_num:
    alpha, beta = torch.Tensor([alpha]), torch.Tensor([beta])
# NB: Nothing to do if both are tensors
```

Reply:
Uh and actually both your and my solutions don't work when given a Variable and a number. I'll think how to fix that

Reply:
Thanks, `.new().fill_()` was exactly what I was looking for!